chameleon/tests/integration_tests.rs

```diff
@@ -0,0 +1,58 @@
+use bitcoin::Network;
+use chameleon::ExchangeRatesProvider;
+use graphql::perro::Error;
+use honey_badger::secrets::{derive_keys, generate_keypair, generate_mnemonic};
+use honey_badger::{Auth, AuthLevel};
+use simplelog::TestLogger;
+use std::env;
+use std::sync::{Arc, Once};
+
+static INIT_LOGGER_ONCE: Once = Once::new();
+
+#[cfg(test)]
+#[ctor::ctor]
+fn init() {
+    INIT_LOGGER_ONCE.call_once(|| {
+        TestLogger::init(simplelog::LevelFilter::Info, simplelog::Config::default()).unwrap();
+    });
+}
+
+#[test]
+fn test_list_currency_codes() {
+    let provider = build_provider();
+    let list = provider.list_currency_codes().unwrap();
+    assert_eq!(list, vec!["CHF", "EUR", "SATS", "USD"]);
+}
+
+#[test]
+fn test_get_exchange_rate() {
+    let provider = build_provider();
+    let rate = provider.query_exchange_rate("EUR".to_string()).unwrap();
+    assert!(1000 < rate);
```

Comment:
Such a bear! I wouldn't want the tests to break when BTC breaks through 100k € 😉